src/metabase/query_processor/middleware/update_used_cards.clj

```diff
@@ -0,0 +1,45 @@
+(ns metabase.query-processor.middleware.update-used-cards
+  (:require
+   [clojure.string :as str]
+   [metabase.lib.metadata :as lib.metadata]
+   [metabase.query-processor.store :as qp.store]
+   [metabase.util.log :as log]
+   #_{:clj-kondo/ignore [:discouraged-namespace]}
```

Comment:
:thinking: 

I guess there's not really a way around this. I was wondering if maybe `update-used-cards!*` should be in `models/card.clj` but that's just indirecting away the problem. Interested in Cam's thoughts on having T2/model code at all in the QP